### PR TITLE
fix favourites reloading on profile change

### PR DESCRIPTION
**Pull request by tamland**
Just another profiles hack. Should fix https://trac.kodi.tv/ticket/17478

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
No objections?

File: xbmc/favourites/FavouritesService.cpp [hunk 1]
```diff
@@ -77,8 +77,16 @@ static bool LoadFromFile(const std::string& strPath, CFileItemList& items)
   return true;
 }
 
-CFavouritesService::CFavouritesService(std::string userDataFolder) : m_userDataFolder(std::move(userDataFolder))
+CFavouritesService::CFavouritesService(std::string userDataFolder)
```
Comment (ksooo) [inline xbmc/favourites/FavouritesService.cpp]: would it make sense to change parameter type to <code>const std::string &</code> as you're no longer using the path-by-value-then-move idiom here? Alternatively, you could use <code>std::move</code> to assign <code>m_userDataFolder</code> in <code>CFavouritesService::ReInit</code>.

**Review by ksooo — COMMENTED** [review]
Just one question/suggestion.

File: xbmc/favourites/FavouritesService.cpp (hunk 1 above)
Comment (tamland) [inline xbmc/favourites/FavouritesService.cpp]: added the move!

Comment (ksooo) [inline xbmc/favourites/FavouritesService.cpp]: yep, imo the better approach of the two alternatives. thx.

**Review by ksooo — APPROVED** [review]
👍

**Pull request merged**